utils/stats.go

```diff
@@ -111,72 +111,107 @@ func FilterAndSaveCommits(API_URL string, LAST_COMMIT_SHA string) (bool, string)
 		fmt.Println("Student username ", student_username)
 		// Checking if commit_author is a registered student or not
-		// TODO: Need to check if student_username is in database or not
-		// If in DB, proceed to check more info about commit
-		// if NOT in DB,  "continue" the loop i.e check the next commit
-
-		// making another API request to get more info about the commit like stats
-		api_url, _ := commits[i]["url"].(string)
-		res, _ := MakeRequest(api_url)
-		resBytes := []byte(res)
-		var commit_info map[string]interface{}
-		_ = json.Unmarshal(resBytes, &commit_info)
-
-		commit_stats_map, _ := commit_info["stats"].(map[string]interface{})
-		lines_added := commit_stats_map["additions"]
-		lines_removed := commit_stats_map["deletions"]
-		fmt.Println("needed_info -> lines-added ", lines_added)     // remove this print later
-		fmt.Println("needed_info -> lines-removed ", lines_removed) // remove this print later
-
-		commit_message := commit_info_map["message"]
-		fmt.Println("needed info -> message ", commit_message)
-
-		// Fetches the tech on which student worked using file names
-		files_arr, _ := commit_info["files"].([]interface{})
-		var file_names []string
-		for j := range files_arr {
-			file_map := files_arr[j].(map[string]interface{})
-			file_name := file_map["filename"].(string)
-			file_names = append(file_names, file_name)
+
+		student := &models.Student{}
+		// CAN BE WRONG
+		db.Where("username=?", student_username).First(&student)
+		fmt.Print(student)
+		if student.ID == 0 {
+			continue
+		} else {
+			// making another API request to get more info about the commit like stats
+			api_url, _ := commits[i]["url"].(string)
+			res, _ := MakeRequest(api_url)
+			resBytes := []byte(res)
+			var commit_info map[string]interface{}
+			_ = json.Unmarshal(resBytes, &commit_info)
+
+			commit_stats_map, _ := commit_info["stats"].(map[string]interface{})
+			lines_added := commit_stats_map["additions"].(float64)
+			lines_removed := commit_stats_map["deletions"].(float64)
+			fmt.Println("needed_info -> lines-added ", lines_added)     // remove this print later
+			fmt.Println("needed_info -> lines-removed ", lines_removed) // remove this print later
+
+			commit_message := commit_info_map["message"].(string)
+			fmt.Println("needed info -> message ", commit_message)
+
+			// Fetches the tech on which student worked using file names
+			files_arr, _ := commit_info["files"].([]interface{})
+			var file_names []string
+			for j := range files_arr {
+				file_map := files_arr[j].(map[string]interface{})
+				file_name := file_map["filename"].(string)
+				file_names = append(file_names, file_name)
+			}
+			languages_worked := GetLanguagesFromFilenames(file_names)
+			fmt.Println("languages worked is ", languages_worked)
+			// TODO: Update the Languages Worked Field under Student row
+			languages_worked=["abcd"]
+			db.Exec("UPDATE students SET tech_worked = ? WHERE username = ?", languages_worked, student_username)
```

Comment:
is There any syntax error here , getting an error msg

![image](https://user-images.githubusercontent.com/68731551/144981220-56e124c6-4351-4c87-8524-ec52ed43e0fd.png)

Reply:
Wrote 175 bcz there is some issue in find tech worked fucntion , and was always returing an empty array, so created an testing array just before updating